google/data_source_dns_managed_zone.go

```diff
@@ -10,11 +16,12 @@ func dataSourceDnsManagedZone() *schema.Resource {
 			"dns_name": &schema.Schema{
 				Type:     schema.TypeString,
 				Computed: true,
+				Optional: true,
 			},
 
 			"name": &schema.Schema{
 				Type:     schema.TypeString,
-				Required: true,
+				Optional: true,
```

Comment:
You may want to take advantage of [`ConflictsWith`](https://godoc.org/github.com/hashicorp/terraform/helper/schema#Schema.ConflictsWith) to ensure both these values can't be set.